libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -247,33 +254,52 @@ class MapboxTripSession(
     }
 
     private fun updateRawLocation(rawLocation: Location) {
+        locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
         ioJobController.scope.launch {
-            rawLocation.let {
-                navigator.updateLocation(it)
-            }
+            val currentDate = Date()
+            lastStatusUpdateTimeMillis = SystemClock.elapsedRealtime()
+            navigator.updateLocation(rawLocation, currentDate)
+            updateDataFromNavigatorStatus(currentDate)
         }
-        locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
+
         if (this.rawLocation == null) {
-            fireOffStatusPolling()
+            ioJobController.scope.launch {
+                while (isActive) {
+                    launch {
+                        emitUnconditionallyGetStatus()
+                    }
+                    delay(STATUS_POLLING_INTERVAL)
```

Comment:
nit: the first open may have 2 get status calls. 
```suggestion
                while (isActive) {
                    delay(STATUS_POLLING_INTERVAL)
                    launch {
                        emitUnconditionallyGetStatus()
                    }
```